src/config/configure.in

```diff
@@ -1720,9 +1720,9 @@ then
    else
       if test "$hypre_using_openmp" = "yes"
       then
-         AC_CHECK_PROGS(CC, [mpxlc mpixlc_r mpixlc mpiicc mpiicx mpigcc mpicc mpipgcc mpipgicc])
+         AC_CHECK_PROGS(CC, [mpicc mpxlc mpixlc_r mpixlc mpiicc mpiicx mpigcc mpipgcc mpipgicc])
```

Comment:
This changes the behavior of hypre's configure at LLNL machines, such as lassen, where xlc is picked by default.

Reply:
`mpicc` is the same as `mpixlc` on Lassen. So this change seems fine to me:
``` 
vicmagri@lassen708 ~ $ mpixlc --version
IBM XL C/C++ for Linux, V16.1.1 (5725-C73, 5765-J13)
Version: 16.01.0001.0014
vicmagri@lassen708 ~ $ mpicc --version
IBM XL C/C++ for Linux, V16.1.1 (5725-C73, 5765-J13)
Version: 16.01.0001.0014
```

The main reason for this change is that `mpicc` is more general, so it makes sense to test for it first.

Reply:
They may not always be the same, under different module loads. If one wants to test mpicc, they can always do CC=mpicc. I think my question is still: why change given that this works fine for long time? :)

Reply:
I think I agree with @liruipeng here that we should not reorder these.  This will change the behavior for many existing users.